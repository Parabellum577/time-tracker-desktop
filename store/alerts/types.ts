import { ActionType } from 'typesafe-actions'

import * as actions from './actions'
import { IError } from '@services/types'

export type AlertsAction = ActionType<typeof actions>

export interface IAlert {
  alertType: 'success' | 'error' | 'warning' | 'coldstart'
  alertMessage: string | IError
}

export interface IAlertsState {
  alertsArray: IAlert[]
}
