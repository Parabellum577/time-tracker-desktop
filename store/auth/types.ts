import { ActionType } from 'typesafe-actions'
import { DeepReadonly } from 'utility-types'

import * as actions from './actions'
import { IError } from '@types'

export type AuthAction = ActionType<typeof actions>

export type IAuthState = DeepReadonly<{
  readonly authorized: boolean
  readonly error?: IError | null
  readonly loading: boolean
  readonly isLoadingData: boolean
}>
