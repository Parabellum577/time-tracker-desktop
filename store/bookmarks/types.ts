import { IFullIssue } from '@types'

export interface IBookmarks {
  extendedBookmarks: IFullIssue[]
}
