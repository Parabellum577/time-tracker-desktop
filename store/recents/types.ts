import { IFullIssue } from '@types'

export interface IRecent {
  extendedRecent: IFullIssue[]
}
