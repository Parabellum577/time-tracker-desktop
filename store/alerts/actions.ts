import { createAsyncAction } from 'typesafe-actions'
import { IError } from '@types'
import { IAlert } from './types'

export const showAlertMessage = createAsyncAction(
  'notifications/SHOW_ALERT_MESSAGE_REQUEST',
  'notifications/SHOW_ALERT_MESSAGE_SUCCESS',
  'notifications/SHOW_ALERT_MESSAGE_FAILURE',
)<IAlert, IAlert, IError>()

export const clearAlertMessage = createAsyncAction(
  'notifications/CLEAR_ALERT_MESSAGE_REQUEST',
  'notifications/CLEAR_ALERT_MESSAGE_SUCCESS',
  'notifications/CLEAR_ALERT_MESSAGE_FAILURE',
)<void, void, IError>()
