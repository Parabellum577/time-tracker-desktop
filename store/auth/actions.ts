import { createAction, createAsyncAction } from 'typesafe-actions'
import { IError } from '@types'
import { ILoginUserPayload, ILoginOTSPayload } from '@services/api-types'

export const loginUser = createAsyncAction('auth/LOGIN_REQUEST', 'auth/LOGIN_SUCCESS', 'auth/LOGIN_FAILURE')<
  ILoginUserPayload,
  void,
  IError
>()

export const loginUserViaOAuth = createAsyncAction(
  'auth/LOGIN_VIA_OAUTH_REQUEST',
  'auth/LOGIN_VIA_OAUTH_SUCCESS',
  'auth/LOGIN_VIA_OAUTH_FAILURE',
)<void, void, IError>()

export const otsLogin = createAsyncAction('auth/OTS_LOGIN_REQUEST', 'auth/OTS_LOGIN_SUCCESS', 'auth/OTS_LOGIN_FAILURE')<
  ILoginOTSPayload,
  ILoginOTSPayload,
  IError
>()

export const loadingDataAfterLogin = createAction('auth/LOADING_DATA_AFTER_LOGIN', resolve => (isLoading: boolean) =>
  resolve(isLoading),
)

export const signOut = createAction('auth/SIGN_OUT', resolve => () => resolve())
