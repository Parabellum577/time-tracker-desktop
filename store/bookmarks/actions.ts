import { createAsyncAction, createAction } from 'typesafe-actions'
import { IError, IFullIssue } from '@types'

export const getBookmarks = createAsyncAction(
  'bookmarks/GET_BOOKMARKS_REQUEST',
  'bookmarks/GET_BOOKMARKS_SUCCESS',
  'bookmarks/GET_BOOKMARKS_FAILURE',
)<void, IFullIssue[], IError>()

export const handleBookmark = createAsyncAction(
  'bookmarks/HANDLE_BOOKMARK_REQUEST',
  'bookmarks/HANDLE_BOOKMARK_SUCCESS',
  'bookmarks/HANDLE_BOOKMARK_FAILURE',
)<IFullIssue, IFullIssue, IError>()

export const clearBookmarks = createAction('plannings/CLEAR_BOOKMARKS', action => () => action())
