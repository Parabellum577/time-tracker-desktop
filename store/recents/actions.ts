import { createAsyncAction, createAction } from 'typesafe-actions'
import { IError, IFullIssue, IPlanning } from '@types'

export const getRecent = createAsyncAction(
  'recent/GET_RECENT_REQUEST',
  'recent/GET_RECENT_SUCCESS',
  'recent/GET_RECENT_FAILURE',
)<void, IFullIssue[], IError>()

export const addRecent = createAsyncAction(
  'recent/ADD_RECENT_REQUEST',
  'recent/ADD_RECENT_SUCCESS',
  'recent/ADD_RECENT_FAILURE',
)<IFullIssue, IFullIssue, IError>()

export const clearRecent = createAction('plannings/CLEAR_RECENT', action => () => action())
