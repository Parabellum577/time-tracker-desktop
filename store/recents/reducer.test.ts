import { ActionType, getType } from 'typesafe-actions'

import * as actions from './actions'
import { IRecent } from './types'
import recent from './reducer'
import { SIMPLE_ISSUE } from '@test-helpers/sample'

describe('[Store] Recent Reducer', () => {
  it('[Success] [Snapshot] Should handle getRecent', () => {
    const action: ActionType<typeof actions> = {
      type: getType(actions.getRecent.success),
      payload: [{ ...SIMPLE_ISSUE }, { ...SIMPLE_ISSUE }],
    }

    const currentState: IRecent = {
      extendedRecent: [{ ...SIMPLE_ISSUE }, { ...SIMPLE_ISSUE }],
    }

    expect(recent(currentState, action)).toMatchSnapshot()
  })

  it('[Failure] Should handle getRecent', () => {
    const action: ActionType<typeof actions> = {
      type: getType(actions.getRecent.failure),
      payload: {
        code: 666,
        message: 'SOME_ERROR',
      },
    }

    const currentState: IRecent = {
      extendedRecent: [],
    }

    expect(recent(currentState, action)).toMatchSnapshot()
  })
})
