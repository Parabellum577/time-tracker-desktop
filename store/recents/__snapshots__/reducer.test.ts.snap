// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Store] Recent Reducer [Failure] Should handle getRecent 1`] = `
Object {
  "extendedRecent": Array [],
}
`;

exports[`[Store] Recent Reducer [Success] [Snapshot] Should handle getRecent 1`] = `
Object {
  "extendedRecent": Array [
    Object {
      "Activity": 0,
      "Done": 0,
      "DueDate": 0,
      "Estimate": 0,
      "ID": "string",
      "IsAssigned": true,
      "Priority": Object {
        "ID": 0,
        "Name": "string",
      },
      "ProjectID": "string",
      "Spent": 0,
      "Status": Object {
        "ID": 0,
        "Name": "string",
      },
      "Title": "string",
      "TrackerID": 0,
      "Type": Object {
        "ID": 0,
        "Name": "string",
      },
      "URL": "string",
    },
    Object {
      "Activity": 0,
      "Done": 0,
      "DueDate": 0,
      "Estimate": 0,
      "ID": "string",
      "IsAssigned": true,
      "Priority": Object {
        "ID": 0,
        "Name": "string",
      },
      "ProjectID": "string",
      "Spent": 0,
      "Status": Object {
        "ID": 0,
        "Name": "string",
      },
      "Title": "string",
      "TrackerID": 0,
      "Type": Object {
        "ID": 0,
        "Name": "string",
      },
      "URL": "string",
    },
  ],
}
`;
